build.gradle.kts

```diff
@@ -0,0 +1,340 @@
+        classpath(Config.Plugins.buildInfo)
+    }
+}
+
+// See https://github.com/gradle/kotlin-dsl/issues/607#issuecomment-375687119
+subprojects { parent!!.path.takeIf { it != rootProject.path }?.let { evaluationDependsOn(it) } }
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+        mavenLocal()
+    }
+
+    // Skip Javadoc generation for Java 1.8 as it breaks build
+    if (JavaVersion.current().isJava8Compatible) {
+        tasks.withType<Javadoc> {
+            options {
+                this as StandardJavadocDocletOptions
+                addStringOption("Xdoclint:none", "-quiet")
+            }
+        }
+    }
+
+    if ((group as String).isNotEmpty() && name != "lint" && name != "internal") {
+        configureAndroid()
+        configureQuality()
+
+        val isLibrary = name == "library"
+        if (Config.submodules.contains(name) || isLibrary) {
+            setupPublishing(isLibrary)
+        }
+    }
+}
+
+val Project.configDir get() = "$rootDir/library/quality"
+val Project.reportsDir get() = "$buildDir/reports"
+
+fun Project.configureAndroid() {
+    apply(plugin = "com.android.${if (name == "app" || name == "proguard-tests") {
+        "application"
+    } else {
+        "library"
+    }}")
+
+    configure<BaseExtension> {
+        compileSdkVersion(Config.SdkVersions.compile)
+
+        defaultConfig {
+            minSdkVersion(Config.SdkVersions.min)
+            targetSdkVersion(Config.SdkVersions.target)
+
+            versionName = Config.version
+            versionCode = 1
+
+            resourcePrefix("fui_")
+            vectorDrawables.useSupportLibrary = true
+        }
+
+        lintOptions {
+            disable(
+                    "ObsoleteLintCustomCheck", // TODO ButterKnife will fix this in v9.0
+                    "IconExpectedSize",
+                    "InvalidPackage", // Firestore uses GRPC which makes lint mad
+                    "NewerVersionAvailable", "GradleDependency", // For reproducible builds
+                    "SelectableText", "SyntheticAccessor" // We almost never care about this
+            )
+
+            isCheckAllWarnings = true
+            isWarningsAsErrors = true
+            isAbortOnError = true
+
+            baselineFile = file("$configDir/lint-baseline.xml")
+            htmlOutput = file("$reportsDir/lint-results.html")
+            xmlOutput = file("$reportsDir/lint-results.xml")
+        }
+    }
+}
+
+fun Project.configureQuality() {
+    apply(plugin = "checkstyle")
+
+    configure<CheckstyleExtension> { toolVersion = "8.10.1" }
+    check { dependsOn("checkstyle") }
+
+    task<Checkstyle>("checkstyle") {
+        configFile = file("$configDir/checkstyle.xml")
+        source("src")
+        include("**/*.java")
+        exclude("**/gen/**")
+        classpath = files()
+    }
+}
+
+fun Project.setupPublishing(isLibrary: Boolean) {
+    val publicationName = "${if (isLibrary) "monolith" else name}Library"
```

Comment:
I'd prefer not to have expressions in strings.  How about:

`val publicationName = if (isLibrary) "monolithLibrary" else "${name}Library"`